crates/api_models/src/customers.rs

```diff
@@ -9,9 +9,8 @@ use crate::payments;
 #[derive(Debug, Default, Clone, Deserialize, Serialize, ToSchema)]
 pub struct CustomerRequest {
     /// The identifier for the customer object. If not provided the customer ID will be autogenerated.
-    #[schema(max_length = 255, example = "cus_y3oqhf46pyzuxjbcn2giaqnb44")]
-    #[serde(default = "generate_customer_id")]
-    pub customer_id: String,
+    #[schema(value_type = Option<String>, max_length = 64, min_length = 1, example = "cus_y3oqhf46pyzuxjbcn2giaqnb44")]
+    pub customer_id: Option<id_type::CustomerId>,
```

Comment:
If not passed in request, we will generate a new customer_id?

Reply:
Yes we are generating the `customer_id` here https://github.com/juspay/hyperswitch/blob/6196afa865adff39b5cddcb062054489957f5191/crates/router/src/core/customers.rs#L42